app/lib/dqt/params/trn_request.rb

```diff
@@ -177,10 +177,11 @@ def start_date
       end
 
       def estimated_course_duration
-        return 3.years if UNDERGRAD_ROUTES.include?(trainee.training_route)
-        return 2.years if trainee.part_time?
+        return trainee.part_time? ? 70.months : 34.months if trainee.provider_led_undergrad?
```

Comment:
I _think_ this works - but I had to really think about whether the ternary or the `if` statement at the end was going to evaluate first!

Maybe if would read more clearly if it was an `if... end` statement with a ternary inside it?

Or even separate one-line `if` statements for each condition? e.g.

```
return 34.months if trainee.provider_led_undergrad && trainee.part_time
```

I'll leave it up to you to decide which looks the best!

Reply:
makes sense - I toyed around with how best to write this for a while and couldn't decide! I might go back to the original guard clauses that I had